### PR TITLE
Merge All Current Cargo Stylus Subcommands Into Main Package

### LINKED ISSUE #83: Change Subcommand from `alias` to `visible_alias`
I didn't realize there were aliases for the subcommands until I looked through the source code. It would be nice to have them visible when using the `--help` option.

Happy to open a PR!

### LINKED ISSUE #88: Add cargo stylus version to the name of created docker image
Currently created docker image name includes only the version of the rust toolchain and does not include the version of the cargo stylus package (`CARGO_PKG_VERSION`) - https://github.com/OffchainLabs/cargo-stylus/blob/main/check/src/docker.rs#L44-L47

That may result in the following scenario when next versions of the cargo stylus are released:
1. A user has used `cargo stylus 0.4.2` with rust-toolchain 1.80.0, deployed some contracts via that setup and has `cargo-stylus-1.80.0` image installed locally.
2. The user updates cargo stylus to v0.5.0
3. Deploys a new contract with new `v0.5.0` version. The code then just checks if `cargo-stylus-1.80.0` image exists locally (https://github.com/OffchainLabs/cargo-stylus/blob/main/check/src/docker.rs#L45) and as it already does, it uses that image that internally has `cargo stylus 0.4.2` installed.
4. When trying to verify the contract, the user believes that v0.5.0 version was used, while in reality it was v0.4.2

### Solution
I suggest to add a current cargo stylus version in the docker image name as well - e.g. `cargo-stylus-0.4.2-1.80.0`

### PR DESCRIPTION
**Pull request by rauljordan**
Currently, we have a weird `cargo-stylus-check`, `cargo-stylus-replay`, and a few other binaries that are then invoked using raw command utilities inside of `cargo-stylus`, which is the main binary. This approach gives us the flexibility of supporting custom extensions to cargo stylus. However, all the current subcommands are arguably critical and should be part of the principal cargo stylus binary. This refactors the repo to do that, while maintaining the flexibility of supporting custom extensions.

Merge #78, #79, #80, #81 first

Closes #83 
Closes #88

### CODE REVIEW

**Review by gligneul — COMMENTED** [review]
This PR looks great. It really simplifies the whole repository. I will do a final review and approve after the merges of the previous PRs.

**Review by rory-ocl — APPROVED** [review]

**Pull request merged**
